anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorMapper.java

```diff
@@ -15,43 +15,95 @@
  */
 package com.expedia.adaptivealerting.anomdetect;
 
+
 import com.expedia.adaptivealerting.anomdetect.comp.DetectorSource;
+import com.expedia.adaptivealerting.anomdetect.mapper.CacheUtil;
+import com.expedia.adaptivealerting.anomdetect.mapper.Detector;
+import com.expedia.adaptivealerting.anomdetect.mapper.DetectorMatchResponse;
 import com.expedia.adaptivealerting.core.data.MappedMetricData;
 import com.expedia.metrics.MetricData;
+import io.micrometer.core.instrument.Metrics;
 import lombok.Getter;
 import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
 import java.util.Set;
-import java.util.stream.Collectors;
-
-import static com.expedia.adaptivealerting.core.util.AssertUtil.notNull;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
  * Entry into the Adaptive Alerting runtime. Its job is find for any incoming {@link MetricData} the corresponding set
  * of mapped detectors, creating a {@link MappedMetricData} for each.
  */
-@RequiredArgsConstructor
 @Slf4j
 public class DetectorMapper {
-
+    private AtomicLong indexSize;
```

Comment:
What is `indexSize`? Consider adding a comment to explain.

From the code it looks like this might be the number of search indices. If so, a better name would be `numSearchIndices`, since `indexSize` sounds like the size of a single index. Also consider updating the gauge name as well (see below).

Reply:
That's something we added for debugging isn't required now